### PR TITLE
Project Beats: add skip controls

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Adds outside of playback skip forward/backwards controls. These advance the playhead forwards or backwards by one measure. Also clicking a measure number or the measures header background will move the playhead to that position outside of playback. To show the current start position, we show the playhead outside of playback in a neutral light color, which switches to the current active yellow when playback starts.

https://github.com/code-dot-org/code-dot-org/assets/85528507/1ab4b981-5df2-4525-98dc-ed55ccc194e8

## Links

[Music Lab Functional Spec](https://docs.google.com/document/d/1vssq4yGvuP2XbI71-drzqNdmNC37pZh46oyqRFee618/edit)

## Testing story

Tested locally.

### CODE REVIEW

**Review by levadadenys — APPROVED** [review]
LGTM!
Wow, the progress with Music Lab is awesome. Looks like web version of Ableton/FL studio to me (which is awesome 😄) already or at least is getting closer and closer to it

**Comment by breville** [comment]
Very cool.  Some initial UI thoughts:

- I really liked the `Run` button being the full width of the panel.  Could we possible put these buttons in the empty space below the triggers?
- With white backgrounds, these buttons feel more prominent than then `Run` button.  Could we tone them down or consider purple?
- We should disable them when playing, since they don't do anything.
- Would it be worth having a third button which jumps the playhead back to the beginning?  
- This exposes the deeper issue that we don't support playing a sample that's not at its beginning.  Most noticeably, if you start from one bar into the song, but the song comprises two-bar long samples (which we use a lot), then you don't hear that sound at all.  Should we consider updating our audio implementation to support starting a sound from partway through?

**Comment by sanchitmalhotra126** [comment]
> * I really liked the `Run` button being the full width of the panel.  Could we possible put these buttons in the empty space below the triggers?

No strong preference here. I went with the [current mockups](https://docs.google.com/document/d/1vssq4yGvuP2XbI71-drzqNdmNC37pZh46oyqRFee618/edit) that have the Run and Skip buttons in one row.

> * With white backgrounds, these buttons feel more prominent than then `Run` button.  Could we tone them down or consider purple?

No strong preference here either. Maybe a more neutral grey?

> * We should disable them when playing, since they don't do anything.

Good call.

> * Would it be worth having a third button which jumps the playhead back to the beginning?

Hmm, yeah that would be helpful. Alternatively, in discussing with @mikeharv another idea could be to have a separate play button that plays from the current location, while "Run" always plays from the beginning?

> * This exposes the deeper issue that we don't support playing a sample that's not at its beginning.  Most noticeably, if you start from one bar into the song, but the song comprises two-bar long samples (which we use a lot), then you don't hear that sound at all.  Should we consider updating our audio implementation to support starting a sound from partway through?

Yep true, though this would be a little trickier to support. I'd defer to @moneppo on how necessary this feels?

**Comment by breville** [comment]
One other thought: should we parameterize the appearance of this feature?  It feels like we might want to introduce it at a later point in a tutorial sequence, when things start to get more complex.

**Comment by breville** [comment]
Two issues we discussed offline:
- after hitting stop, the highlighted Blockly blocks don't unhighlight.  
- for the disabled buttons, we can try making the existing ones `opacity: 50%`.

**Comment by sanchitmalhotra126** [comment]
@breville revisiting this PR, addressing the feedback we discussed

> * after hitting stop, the highlighted Blockly blocks don't unhighlight.

this should be fixed now!

> * for the disabled buttons, we can try making the existing ones `opacity: 50%`.

Done

>One other thought: should we parameterize the appearance of this feature? It feels like we might want to introduce it at a later point in a tutorial sequence, when things start to get more complex.

This feature is now hidden behind a config flag (`skip-controls=(enabled|disabled)`). I've also gone ahead and put keyboard shortcuts behind a config flag as well (`keyboard-shortcuts=(enabled|disabled)`). Both will default to disabled if not specified.

File: apps/src/musicMenu/MusicMenu.jsx [hunk 1]
```diff
@@ -82,6 +82,22 @@ const optionsList = [
     type: 'string',
     description: 'Set a specific key by name (i.e. "C", "C#", "D", etc)',
   },
+  {
+    name: 'skip-controls',
+    type: 'radio',
+    values: [
+      {value: 'disabled', description: 'Disable skip controls.'},
+      {value: 'enabled', description: 'Enable skip controls.'},
```
Comment (breville) [inline apps/src/musicMenu/MusicMenu.jsx]: super nit: we use `true` and `false` for other values in the menu.  `enabled` and `disabled` are definitely nice, but should we be consistent with whichever we choose?  One factor might be what works best in the context of levelbuilder parameters.

**Review by breville — APPROVED** [review]
Great feature!

File: apps/src/musicMenu/MusicMenu.jsx (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/musicMenu/MusicMenu.jsx]: Oh good call, I'll make this `true`/`false` for consistency.

**Pull request merged**